pkg/autoscaler/metrics/collector.go

```diff
@@ -134,27 +134,17 @@ func NewMetricCollector(statsScraperFactory StatsScraperFactory, logger *zap.Sug
 
 // CreateOrUpdate either creates a collection for the given metric or update it, should
 // it already exist.
-// Map access optimized via double-checked locking.
 func (c *MetricCollector) CreateOrUpdate(metric *av1alpha1.Metric) error {
 	scraper, err := c.statsScraperFactory(metric, c.logger)
 	if err != nil {
 		return err
 	}
 	key := types.NamespacedName{Namespace: metric.Namespace, Name: metric.Name}
 
-	c.collectionsMutex.RLock()
```

Comment:
Before y'all yell at me: The maximum concurrency at this piece of code is 2. I think I only put this in here because doubly-checked locking was my favorite toy at the time. :sweat_smile:

Reply:
😂